gems/pending/util/extensions/miq-file.rb

```diff
@@ -53,12 +53,17 @@ def self.sizeEx(path)
 
   def self.path_to_uri(file, hostname = nil)
     hostname ||= MiqSockUtil.getFullyQualifiedDomainName
+    file = Addressable::URI.encode(file.tr('\\', '/'))
 
-    URI::Generic.build(
-      :scheme => 'file',
-      :host   => hostname,
-      :path   => "/#{Addressable::URI.encode(file.tr('\\', '/'))}"
-    ).to_s
+    begin
+      URI::Generic.build(
+        :scheme => 'file',
+        :host   => hostname,
+        :path   => "/#{file}"
+      ).to_s
+    rescue URI::InvalidComponentError # Punt on Windows volume names, etc
+      "file://#{hostname}/#{file}"
```

Comment:
Not that I can tell. The only error I can get it to raise in my (limited) testing is `URI::InvalidComponentError: bad component(expected absolute path component)`. It's really just doing a regex check against an ABS_PATH constant in a file called uri/rfc3986_parser.rb.

This method is only used in one place by the infra manager refresh parser, and the information is just plain text displayed within the UI. So, in the unlikely occurrence of a mistake, it won't have any real impact.